.github/workflows/push.yml

```diff
@@ -0,0 +1,33 @@
+name: Push
+on:
+  pull_request:
+    branches:
+      - master
+jobs:
+  test:
+    name: image
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+    defaults:
+      run:
+        working-directory: ./src/github.com/${{ github.repository }}
+    env:
+      GOPATH: ${{ github.workspace }}
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+        with:
+          # check out actual commit that triggered event
+          ref: ${{ github.event.pull_request.head.sha }}
+          path: ./src/github.com/${{ github.repository }}
+      - name: Login to DockerHub
+        uses: docker/login-action@v1
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_PASSWORD }}
+      - name: Build
+        run: |
+          make latest && make versioned
+      - name: Push image
+        run: |
+          make push-versioned
```

Comment:
We should run tests prior to pushing an image. There are ways to kick off actions from other actions, somehow. Easier way may be to combine push in the pr.yml after tests are run.

Reply:
Drew, tests will start in other action (it triggers on each new commit). Please take a look second file in that PR

Reply:
The actions run independently though, so even if the tests fail the push action could still work, couldn't it?

Reply:
oh, sorry.. that's right. probably I had to make single file with direct dependency (like **needs** keyword)

Reply:
I pushed a commit to make it just run the tests for both PR and Push. It seems logical to me that they would be separate actions, but as I noted, I'm not a GitHub Actions expert.